### PR TITLE
Consolidate Noobaa and Ceph Plugin

### PR DESCRIPTION
**Pull request by bipuladh**
After: 
```
./packages/ceph-storage-plugin/src/
├── components
│   ├── bucket-class
│   │   ├── _backingstore-table.scss
│   │   ├── backingstore-table.tsx
│   │   ├── create-bc.scss
│   │   ├── create-bc.tsx
│   │   ├── modals
│   │   │   ├── _bs-modal.scss
│   │   │   └── edit-backingstore-modal.tsx
│   │   ├── state.ts
│   │   └── wizard-pages
│   │       ├── backingstore-page.tsx
│   │       ├── general-page.tsx
│   │       ├── placement-policy-page.tsx
│   │       └── review-page.tsx
│   ├── create-backingstore-page
│   │   ├── create-bs-modal.tsx
│   │   ├── create-bs-page.tsx
│   │   ├── create-bs.scss
│   │   └── create-bs.tsx
│   ├── dashboards
│   │   ├── common
│   │   │   ├── breakdown-card
│   │   │   │   ├── breakdown-body.tsx
│   │   │   │   ├── breakdown-capacity.tsx
│   │   │   │   ├── breakdown-card.scss
│   │   │   │   ├── breakdown-chart.tsx
│   │   │   │   ├── breakdown-dropdown.tsx
│   │   │   │   ├── breakdown-loading.tsx
│   │   │   │   ├── consts.ts
│   │   │   │   └── utils.tsx
│   │   │   ├── capacity-breakdown
│   │   │   │   ├── breakdown-body.tsx
│   │   │   │   ├── breakdown-capacity.tsx
│   │   │   │   ├── breakdown-card.scss
│   │   │   │   ├── breakdown-chart.tsx
│   │   │   │   ├── breakdown-dropdown.tsx
│   │   │   │   ├── breakdown-loading.tsx
│   │   │   │   ├── consts.ts
│   │   │   │   └── utils.tsx
│   │   │   ├── data-resiliency
│   │   │   │   └── data-resiliency-activity.tsx
│   │   │   └── storage-efficiency
│   │   │       ├── storage-efficiency-card-item.tsx
│   │   │       └── storage-efficiency-card.scss
│   │   ├── object-service
│   │   │   ├── activity-card
│   │   │   │   ├── activity-card.scss
│   │   │   │   ├── activity-card.tsx
│   │   │   │   └── data-resiliency-activity
│   │   │   │       ├── data-resiliency-activity.scss
│   │   │   │       └── data-resiliency-activity.tsx
│   │   │   ├── buckets-card
│   │   │   │   ├── buckets-card-item.tsx
│   │   │   │   ├── buckets-card.scss
│   │   │   │   ├── buckets-card.tsx
│   │   │   │   └── utils.ts
│   │   │   ├── capacity-breakdown
│   │   │   │   ├── capacity-breakdown-card.scss
│   │   │   │   └── capacity-breakdown-card.tsx
│   │   │   ├── data-consumption-card
│   │   │   │   ├── data-consumption-card-dropdown.tsx
│   │   │   │   ├── data-consumption-card.scss
│   │   │   │   ├── data-consumption-card.tsx
│   │   │   │   ├── data-consumption-card-utils.ts
│   │   │   │   ├── data-consumption-graph.tsx
│   │   │   │   └── performance-graph.tsx
│   │   │   ├── details-card
│   │   │   │   ├── details-card.scss
│   │   │   │   └── details-card.tsx
│   │   │   ├── resource-providers-card
│   │   │   │   ├── resource-providers-card-body.tsx
│   │   │   │   ├── resource-providers-card-item.tsx
│   │   │   │   ├── resource-providers-card.scss
│   │   │   │   └── resource-providers-card.tsx
│   │   │   ├── status-card
│   │   │   │   ├── object-service-health.tsx
│   │   │   │   ├── status-card.scss
│   │   │   │   ├── status-card.tsx
│   │   │   │   └── statuses.tsx
│   │   │   └── storage-efficiency-card
│   │   │       └── storage-efficiency-card.tsx
│   │   ├── persistent-external
│   │   │   ├── breakdown-card.tsx
│   │   │   ├── details-card.tsx
│   │   │   ├── status-card.tsx
│   │   │   └── utilization-card.tsx
│   │   └── persistent-internal
│   │       ├── activity-card
│   │       │   ├── activity-card.scss
│   │       │   ├── activity-card.tsx
│   │       │   ├── cluster-expand-activity.tsx
│   │       │   └── ocs-upgrade-activity.tsx
│   │       ├── capacity-breakdown-card
│   │       │   ├── capacity-breakdown-card.scss
│   │       │   └── capacity-breakdown-card.tsx
│   │       ├── details-card.tsx
│   │       ├── inventory-card.tsx
│   │       ├── raw-capacity-card
│   │       │   ├── raw-capacity-card.scss
│   │       │   └── raw-capacity-card.tsx
│   │       ├── status-card
│   │       │   ├── status-card.tsx
│   │       │   └── utils.ts
│   │       ├── storage-efficiency-card
│   │       │   └── storage-efficiency-card.tsx
│   │       ├── top-consumers-card
│   │       │   ├── top-consumers-card-body.tsx
│   │       │   └── utils.ts
│   │       └── utilization-card
│   │           ├── utilization-card.tsx
│   │           └── utils.tsx
│   ├── disk-inventory
│   │   ├── ocs-disks-list.tsx
│   │   ├── ocs-kebab-options.tsx
│   │   ├── ocs-status-column.scss
│   │   ├── ocs-status-column.tsx
│   │   └── state-reducer.ts
│   ├── kms-config
│   │   ├── kms-config.scss
│   │   ├── kms-config.tsx
│   │   └── utils.tsx
│   ├── modals
│   │   ├── add-capacity-modal
│   │   │   ├── add-capacity-modal.scss
│   │   │   └── add-capacity-modal.tsx
│   │   ├── advanced-kms-modal
│   │   │   ├── advanced-kms-modal.scss
│   │   │   └── advanced-kms-modal.tsx
│   │   ├── disk-replacement-modal.tsx
│   │   ├── storage-class-dropdown.scss
│   │   ├── storage-class-dropdown.tsx
│   │   └── storage-pool-modal
│   │       ├── storage-pool-modal.scss
│   │       └── storage-pool-modal.tsx
│   ├── object-bucket-claim-page
│   │   ├── create-obc.scss
│   │   ├── create-obc.tsx
│   │   ├── menu-actions.ts
│   │   ├── modals
│   │   │   └── attach-deployment-obc-modal.tsx
│   │   ├── object-bucket-claim.tsx
│   │   └── secret.tsx
│   ├── object-bucket-page
│   │   ├── create-ob.tsx
│   │   ├── object-bucket.tsx
│   │   └── state.ts
│   ├── ocs-install
│   │   ├── attached-devices-mode
│   │   │   ├── attached-devices.scss
│   │   │   ├── create-sc
│   │   │   │   ├── create-sc.tsx
│   │   │   │   ├── state.ts
│   │   │   │   └── wizard-pages
│   │   │   │       ├── auto-detect-volume.tsx
│   │   │   │       ├── configure-step.tsx
│   │   │   │       ├── create-local-volume-set.tsx
│   │   │   │       ├── disk-list.tsx
│   │   │   │       ├── donut-chart.tsx
│   │   │   │       ├── review-and-create-step.tsx
│   │   │   │       └── storage-and-nodes-step.tsx
│   │   │   ├── install.tsx
│   │   │   └── sc-node-list.tsx
│   │   ├── existing-cluster-modal.tsx
│   │   ├── external-mode
│   │   │   ├── fileUpload.scss
│   │   │   ├── fileUpload.tsx
│   │   │   ├── install.scss
│   │   │   ├── install.tsx
│   │   │   ├── utils.spec.ts
│   │   │   └── utils.ts
│   │   ├── install-page.scss
│   │   ├── install-page.tsx
│   │   ├── install-wizard
│   │   │   ├── capacity-and-nodes.tsx
│   │   │   ├── _configure.scss
│   │   │   ├── configure.tsx
│   │   │   ├── install-wizard.scss
│   │   │   └── review-and-create.tsx
│   │   ├── internal-mode
│   │   │   ├── install-wizard-steps
│   │   │   │   ├── configure.tsx
│   │   │   │   ├── index.ts
│   │   │   │   ├── review-and-create.tsx
│   │   │   │   └── select-capacity-and-nodes-step.tsx
│   │   │   ├── install-wizard.tsx
│   │   │   └── reducer.ts
│   │   ├── node-list.tsx
│   │   ├── ocs-install.scss
│   │   ├── ocs-request-data.ts
│   │   ├── pvs-available-capacity.scss
│   │   ├── pvs-available-capacity.tsx
│   │   ├── subscription-icon.scss
│   │   ├── subscription-icon.tsx
│   │   └── types.ts
│   └── ocs-storage-class-form
│       ├── ocs-storage-class-form.scss
│       └── ocs-storage-class-form.tsx
├── constants
│   ├── capacity-breakdown.ts
│   ├── common.ts
│   ├── data-consumption.ts
│   ├── disks.ts
│   ├── index.ts
│   ├── kms.ts
│   ├── network.ts
│   ├── ocs-install.ts
│   ├── pool.ts
│   ├── providers.ts
│   ├── status.ts
│   ├── storage-pool-const.ts
│   └── tooltips.ts
├── extensions
│   └── dashboards.ts
├── features.ts
├── index.ts
├── __mocks__
│   ├── breakdown-data.ts
│   └── independent-mode-dashboard-data.ts
├── models.ts
├── plugin.ts
├── queries
│   ├── ceph-queries.ts
│   ├── index.ts
│   └── object-storage-queries.ts
├── resources.ts
├── selectors
│   └── index.ts
├── table-filters.ts
├── __tests__
│   ├── breakdown-body.spec.tsx
│   ├── breakdown-chart.spec.tsx
│   ├── independent-dashboard-breakdown-card.spec.tsx
│   ├── independent-dashboard-details-card.spec.tsx
│   └── __snapshots__
├── types.ts
└── utils
    ├── add-capacity.ts
    ├── alert-action-path.tsx
    ├── common-ocs-install-el.tsx
    ├── common.ts
    ├── dashboard.ts
    ├── events.ts
    ├── index.ts
    ├── install.ts
    ├── noobaa-utils.ts
    ├── ocs-storage-class-params.tsx
    ├── osd-size-dropdown.tsx
    └── storage-pool.tsx
```

Before: 
```
./packages/ceph-storage-plugin/src/
├── components
│   ├── dashboards
│   │   ├── common
│   │   │   ├── capacity-breakdown
│   │   │   │   ├── breakdown-body.tsx
│   │   │   │   ├── breakdown-capacity.tsx
│   │   │   │   ├── breakdown-card.scss
│   │   │   │   ├── breakdown-chart.tsx
│   │   │   │   ├── breakdown-dropdown.tsx
│   │   │   │   ├── breakdown-loading.tsx
│   │   │   │   ├── consts.ts
│   │   │   │   └── utils.tsx
│   │   │   ├── data-resiliency
│   │   │   │   └── data-resiliency-activity.tsx
│   │   │   └── storage-efficiency
│   │   │       ├── storage-efficiency-card-item.tsx
│   │   │       └── storage-efficiency-card.scss
│   │   ├── persistent-external
│   │   │   ├── breakdown-card.tsx
│   │   │   ├── details-card.tsx
│   │   │   ├── status-card.tsx
│   │   │   └── utilization-card.tsx
│   │   └── persistent-internal
│   │       ├── activity-card
│   │       │   ├── activity-card.scss
│   │       │   ├── activity-card.tsx
│   │       │   ├── cluster-expand-activity.tsx
│   │       │   └── ocs-upgrade-activity.tsx
│   │       ├── capacity-breakdown-card
│   │       │   ├── capacity-breakdown-card.scss
│   │       │   └── capacity-breakdown-card.tsx
│   │       ├── details-card.tsx
│   │       ├── inventory-card.tsx
│   │       ├── raw-capacity-card
│   │       │   ├── raw-capacity-card.scss
│   │       │   └── raw-capacity-card.tsx
│   │       ├── status-card
│   │       │   ├── status-card.tsx
│   │       │   └── utils.ts
│   │       ├── storage-efficiency-card
│   │       │   └── storage-efficiency-card.tsx
│   │       ├── top-consumers-card
│   │       │   ├── top-consumers-card-body.tsx
│   │       │   └── utils.ts
│   │       └── utilization-card
│   │           ├── utilization-card.tsx
│   │           └── utils.tsx
│   ├── disk-inventory
│   │   ├── ocs-disks-list.tsx
│   │   ├── ocs-kebab-options.tsx
│   │   ├── ocs-status-column.scss
│   │   ├── ocs-status-column.tsx
│   │   └── state-reducer.ts
│   ├── kms-config
│   │   ├── kms-config.scss
│   │   ├── kms-config.tsx
│   │   └── utils.tsx
│   ├── modals
│   │   ├── add-capacity-modal
│   │   │   ├── add-capacity-modal.scss
│   │   │   └── add-capacity-modal.tsx
│   │   ├── advanced-kms-modal
│   │   │   ├── advanced-kms-modal.scss
│   │   │   └── advanced-kms-modal.tsx
│   │   ├── disk-replacement-modal.tsx
│   │   ├── storage-class-dropdown.scss
│   │   ├── storage-class-dropdown.tsx
│   │   └── storage-pool-modal
│   │       ├── storage-pool-modal.scss
│   │       └── storage-pool-modal.tsx
│   ├── ocs-install
│   │   ├── attached-devices-mode
│   │   │   ├── attached-devices.scss
│   │   │   ├── create-sc
│   │   │   │   ├── create-sc.tsx
│   │   │   │   ├── state.ts
│   │   │   │   └── wizard-pages
│   │   │   │       ├── auto-detect-volume.tsx
│   │   │   │       ├── configure-step.tsx
│   │   │   │       ├── create-local-volume-set.tsx
│   │   │   │       ├── disk-list.tsx
│   │   │   │       ├── donut-chart.tsx
│   │   │   │       ├── review-and-create-step.tsx
│   │   │   │       └── storage-and-nodes-step.tsx
│   │   │   ├── install.tsx
│   │   │   └── sc-node-list.tsx
│   │   ├── existing-cluster-modal.tsx
│   │   ├── external-mode
│   │   │   ├── fileUpload.scss
│   │   │   ├── fileUpload.tsx
│   │   │   ├── install.scss
│   │   │   ├── install.tsx
│   │   │   ├── utils.spec.ts
│   │   │   └── utils.ts
│   │   ├── install-page.scss
│   │   ├── install-page.tsx
│   │   ├── install-wizard
│   │   │   ├── capacity-and-nodes.tsx
│   │   │   ├── _configure.scss
│   │   │   ├── configure.tsx
│   │   │   ├── install-wizard.scss
│   │   │   └── review-and-create.tsx
│   │   ├── internal-mode
│   │   │   ├── install-wizard-steps
│   │   │   │   ├── configure.tsx
│   │   │   │   ├── index.ts
│   │   │   │   ├── review-and-create.tsx
│   │   │   │   └── select-capacity-and-nodes-step.tsx
│   │   │   ├── install-wizard.tsx
│   │   │   └── reducer.ts
│   │   ├── node-list.tsx
│   │   ├── ocs-install.scss
│   │   ├── ocs-request-data.ts
│   │   ├── pvs-available-capacity.scss
│   │   ├── pvs-available-capacity.tsx
│   │   ├── subscription-icon.scss
│   │   ├── subscription-icon.tsx
│   │   └── types.ts
│   └── ocs-storage-class-form
│       ├── ocs-storage-class-form.scss
│       └── ocs-storage-class-form.tsx
├── constants
│   ├── common.ts
│   ├── disks.ts
│   ├── index.ts
│   ├── kms.ts
│   ├── network.ts
│   ├── ocs-install.ts
│   ├── pool.ts
│   ├── queries.ts
│   ├── resources.ts
│   ├── storage-pool-const.ts
│   └── tooltips.ts
├── extensions
│   └── dashboards.ts
├── features.ts
├── index.ts
├── __mocks__
│   ├── breakdown-data.ts
│   └── independent-mode-dashboard-data.ts
├── models.ts
├── plugin.ts
├── selectors
│   └── index.ts
├── __tests__
│   ├── breakdown-body.spec.tsx
│   ├── breakdown-chart.spec.tsx
│   ├── independent-dashboard-breakdown-card.spec.tsx
│   ├── independent-dashboard-details-card.spec.tsx
│   └── __snapshots__
├── types.ts
└── utils
    ├── add-capacity.ts
    ├── alert-action-path.tsx
    ├── common-ocs-install-el.tsx
    ├── common.ts
    ├── dashboard.ts
    ├── events.ts
    ├── index.ts
    ├── install.ts
    ├── ocs-storage-class-params.tsx
    ├── osd-size-dropdown.tsx
    └── storage-pool.tsx

```

### CODE REVIEW

**Review by afreen23 — CHANGES_REQUESTED** [review]
There are some stale files which we renamed in #8111  are getting added as new files
Please remove them: 
- breakdown-card
- lso-disk-inventory

**Comment by afreen23** [comment]
1. Modals can be moved to the `modals` dir to be consistent with ceph package conventions:
 ```
├── bucket-class
 ....
│   │   ├── modals
│   │   │   ├── _bs-modal.scss
│   │   │   └── edit-backingstore-modal.tsx
----

│   ├── object-bucket-claim-page
.....
│   │   ├── modals
│   │   │   └── attach-deployment-obc-modal.tsx
```

2. Pointing out (not necessary for this PR) , there is one `types.ts` in `ocs-install` and we  have one for whole package.
```
│   │   ├── subscription-icon.scss
│   │   ├── subscription-icon.tsx
│   │   └── types.ts
```

3. Can we get rid of `index.ts` ?
```
├── queries
│   ├── ceph-queries.ts
│   ├── index.ts
│   └── object-storage-queries.ts
```
I think we want to keep naming to be:
```
│   ├── ceph-queries.ts
│   ├── index.ts
│   └── noobaa-queries.ts
```
If we go with current I would suggest to move RGW queries to `object-storage-queries`


4. `table-filters.ts` can be moved to respective ob/obc dir or be included in the details page itself.

**Comment by bipuladh** [comment]
Thanks for the review @afreen23 
1) Done
2) Let's refactor that in a different PR( let's track migration with this).
3) Same as (2) let's take that in a different PR(RGW queries). Index file makes accessing queries much easier(need to make sure we are not introducing cycles though). 
4) Done

File: frontend/packages/ceph-storage-plugin/locales/en/noobaa-storage-plugin.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "my-multi-cloud-mirror": "my-multi-cloud-mirror",
+  "Tier 1: {{tier1Policy}}": "Tier 1: {{tier1Policy}}",
+  "Tier 2:": "Tier 2:"
+}
```
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/locales/en/noobaa-storage-plugin.json:5]: can you fix this.

**Review by cloudbehl — COMMENTED** [review]
nit: Otherwise LGTM

**Review by afreen23 — APPROVED** [review]
We can change ns name with later branding chnages as well n 4.9

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/common/breakdown-card/breakdown-card.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+.capacity-breakdown-card__capacity-body {
+  font-size: var(--pf-global--FontSize--sm);
+  margin-bottom: 0;
+}
+
+.capacity-breakdown-card__header-link {
+  padding: var(--pf-global--spacer--xs);
+  margin-right: var(--pf-global--spacer--md);
```
Comment (jhadvig) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/common/breakdown-card/breakdown-card.scss:15]: nit: properties should be ordered alphabetically

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 3]
```diff
@@ -457,6 +467,279 @@ const plugin: Plugin<ConsumedExtensions> = [
...
+      path: `/k8s/ns/:ns/${ClusterServiceVersionModel.plural}/:appName/${referenceForModel(
+        models.NooBaaBucketClassModel,
+      )}/~new`,
+      loader: () =>
+        import('./components/bucket-class/create-bc' /* webpackChunkName: "create-bc" */).then(
+          (m) => m.default,
+        ),
+    },
+    flags: {
+      required: [NOOBAA_FLAG],
```
Comment (jhadvig) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts:484]: Shouldn't this be `CEPH_FLAG` ? or is this nooba specific ?

**Review by jhadvig — COMMENTED** [review]
One nit and a question. Otherwise looks good :)

File: frontend/packages/ceph-storage-plugin/src/plugin.ts (hunk 3 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/plugin.ts:484]: This is noobaa specific.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/common/breakdown-card/breakdown-card.scss (hunk 2 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/common/breakdown-card/breakdown-card.scss:15]: There are a lot of places where this needs to be done. We are doing some effort from our side to come up with linter rules. Ex: https://github.com/openshift/console/pull/8131
We can take up ordering of CSS properties in the future.

**Pull request merged**